lib/membrane_http_adaptive_stream/stream_sink_bin.ex

```diff
@@ -0,0 +1,139 @@
+
+  def_options muxer_segment_duration: [
+                spec: pos_integer,
+                default: 2 |> Time.seconds()
+              ],
+              manifest_name: [
+                spec: String.t(),
+                default: "index",
+                description: "Name of the main manifest file"
+              ],
+              manifest_module: [
+                spec: module,
+                description: """
+                Implementation of the `Membrane.HTTPAdaptiveStream.Manifest`
+                behaviour.
+                """
+              ],
+              storage: [
+                spec: Storage.config_t(),
+                description: """
+                Storage configuration. May be one of `Membrane.HTTPAdaptiveStream.Storages.*`.
+                See `Membrane.HTTPAdaptiveStream.Storage` behaviour.
+                """
+              ],
+              target_window_duration: [
+                spec: pos_integer | :infinity,
+                default: Time.seconds(40),
+                description: """
+                Manifest duration is keept above that time, while the oldest segments
+                are removed whenever possible.
+                """
+              ],
+              persist?: [
+                spec: boolean,
+                default: false,
+                description: """
+                If true, stale segments are removed from the manifest only. Once
+                playback finishes, they are put back into the manifest.
+                """
+              ],
+              target_segment_duration: [
+                spec: pos_integer,
+                default: 0,
+                description: """
+                Expected length of each segment. Setting it is not necessary, but
+                may help players achieve better UX.
+                """
+              ]
+
+  def_input_pad :input,
+    demand_unit: :buffers,
+    caps: [Membrane.H264, Membrane.AAC],
+    availability: :on_request,
+    options: [
+      encoding: [
+        spec: :H264 | :AAC,
+        description: """
+        Encoding type determining which payloader will be used for the given stream.
+        """
+      ]
+    ]
+
+  @impl true
+  def handle_init(opts) do
+    children = [
+      sink: %Sink{
+        manifest_name: opts.manifest_name,
+        manifest_module: opts.manifest_module,
+        storage: opts.storage,
+        target_window_duration: opts.target_window_duration,
+        persist?: opts.persist?,
+        target_segment_duration: opts.target_segment_duration
+      }
+    ]
+
+    state = %{muxer_segment_duration: opts.muxer_segment_duration}
+
+    {{:ok, spec: %ParentSpec{children: children}}, state}
+  end
+
+  @impl true
+  def handle_pad_added(Pad.ref(:input, ref) = pad, context, state) do
+    muxer = %MP4.CMAF.Muxer{segment_duration: state.muxer_segment_duration}
+
+    payloader =
+      case context.options[:encoding] do
+        :H264 ->
```

Comment:
You could create a map as a module attribute that wil map those atoms to those payloaders and then do something like `Map.fetch!(@payloaders, context.options.encoding)`